### PR TITLE
adding port and exposed fields to component_types

### PR DESCRIPTION
**Pull request by tinakurian**
Adding port and exposed fields to component_types.

### CODE REVIEW

**Review by sbose78 — APPROVED** [review]
Thanks!

File: pkg/apis/devconsole/v1alpha1/component_types.go [hunk 1]
```diff
@@ -11,6 +11,10 @@ type ComponentSpec struct {
 	BuildType string `json:"buildType"`
 	// GitSourceRef is the source code of your component. Currently, only public remote URLs are supported.
 	GitSourceRef string `json:"gitSourceRef"`
+	// The cluster port of the service for your deployed component. The same port also matches target port.
+	Port int32 `json:"port,omitempty"`
```
Comment (baijum) [inline pkg/apis/devconsole/v1alpha1/component_types.go:15]: Why `omitempty`?

Comment (sbose78) [reply pkg/apis/devconsole/v1alpha1/component_types.go:15]: @corinnekrych can confirm, if that field is absent, we should skip the creation of the `DeploymentConfig`/`Service` . We only create the `BuildConfig` and `ImageStream`

Comment (tinakurian) [reply pkg/apis/devconsole/v1alpha1/component_types.go:15]: Probably a better question for @corinnekrych as she was the one who worked on the story to add this: https://jira.coreos.com/browse/ODC-382

I'm working on tech debt to refactor the code to use component_types.go in devconsole-api and remove it from devconsole-operator

**Pull request merged**